tool_scoring.R

```diff
@@ -0,0 +1,458 @@
+      fields <- c("insertName", "insertSpecies", "vectorType", "insertEntrezId", "vectorBackbone")
+    } else if (tool_type == "Biobank") {
+      # 5 fields with high completion: specimenTissueType, diseaseType, tumorType, specimenFormat, specimenType
+      fields <- c("specimenTissueType", "diseaseType", "tumorType", "specimenFormat", "specimenType")
+    } else {
+      fields <- c()
+    }
+
+    # Count how many critical info fields are filled (evenly distributed to 30 points)
+    if (length(fields) > 0) {
+      for (field in fields) {
+        is_filled <- !is.null(resource_data[[field]]) && !is.na(resource_data[[field]]) &&
+          resource_data[[field]] != "" && resource_data[[field]] != "NULL"
+        if (!is_filled) {
+          critical_info_missing <- c(critical_info_missing, field)
+        }
+      }
+      filled_count <- length(fields) - length(critical_info_missing)
+      critical_info_score <- (filled_count / length(fields)) * 30
+    }
+  }
+
+  missing_fields$critical_info <- paste(critical_info_missing, collapse="; ")
+  score_breakdown$critical_info <- round(critical_info_score, 1)
+  total_score <- total_score + critical_info_score
+
+  # Other info (20 points distributed evenly)
+  other_info_score <- 0
+  other_info_missing <- c()
+
+  if (!is.na(tool_type)) {
+    if (tool_type == "Animal Model") {
+      # 3 fields with medium completion: backgroundSubstrain, synonyms, animalModelOfManifestation
+      fields <- c("backgroundSubstrain", "synonyms", "animalModelOfManifestation")
+    } else if (tool_type == "Cell Line") {
+      # 1 field with medium completion: tissue
+      fields <- c("tissue")
+    } else if (tool_type == "Antibody") {
+      # 1 field with medium completion: cloneId
+      fields <- c("cloneId")
+    } else if (tool_type == "Genetic Reagent") {
+      # 2 fields with medium completion: synonyms, promoter
+      fields <- c("synonyms", "promoter")
+    } else if (tool_type == "Biobank") {
+      # 1 field with medium completion: specimenPreparationMethod
+      fields <- c("specimenPreparationMethod")
+    } else {
+      fields <- c()
+    }
+
+    # Count how many other info fields are filled (evenly distributed to 20 points)
+    if (length(fields) > 0) {
+      for (field in fields) {
+        is_filled <- !is.null(resource_data[[field]]) && !is.na(resource_data[[field]]) &&
+          resource_data[[field]] != "" && resource_data[[field]] != "NULL"
+        if (!is_filled) {
+          other_info_missing <- c(other_info_missing, field)
+        }
+      }
+      filled_count <- length(fields) - length(other_info_missing)
+      other_info_score <- (filled_count / length(fields)) * 20
+    }
+  }
+
+  missing_fields$other_info <- paste(other_info_missing, collapse="; ")
+  score_breakdown$other_info <- round(other_info_score, 1)
+  total_score <- total_score + other_info_score
+
+  # Observations (10 points max)
```

Comment:
After seeing the actual results in your table, I think i'd like to weigh observations a bit more heavily, say, 25-30 points; the current top scorers are things that have lots of details but few observations